tests/TTCacheTest.php

```diff
@@ -719,4 +719,38 @@ protected function hash(string $key): string
     {
         return md5($key);
     }
+
+    /**
+     * @test
+     */
+    function adding_tags_doesnt_invalidate()
+    {
+        $v = $this->tt->remember('testkey', null, [], fn () => 'hello 1');
+        $this->assertEquals('hello 1', $v->value());
+        // If we use the same key but a different callback that returns something different, we should still get
+        // the previously cached value.
+        $v = $this->tt->remember('testkey', null, ['foobar'], fn () => 'hello 2');
+        $this->assertTrue($v->isHit());
+        $this->assertEquals('hello 1', $v->value());
+    }
+
+    /**
+     * @test
+     * @group debug
+     */
+    function invalidating_a_tag_added_invalidates()
+    {
+        $v = $this->tt->remember('testkey', null, ['baz'], fn () => 'hello 1');
+        $this->assertEquals('hello 1', $v->value());
+        // If we use the same key but a different callback that returns something different, we should still get
+        // the previously cached value.
+        $v = $this->tt->remember('testkey', null, ['baz', 'foobar'], fn () => 'hello 2');
```

Comment:
If `foobar` was a pre-existing tag here, we'd get `hello 2` though right? Because it would already have a hash?

Reply:
I'm not sure I follow. If you meant `foobar` being always _here_ then it should still be `hello 1`. If you meant `foobar` is used somewhere else in a different tree then I think should still be `hello 1` (because the cache-key didn't change, we are simply categorizing this `remember(...)` call under an existing tag)

Reply:
Right, I understand now.

Mind if I sit on this one a bit? Want to think through implications and concurrency scenarios before merging